src/lib/queue.js

```diff
@@ -61,6 +61,21 @@ const {
 
 export { generateRedisConfig };
 
+export async function increaseListeners(queue, num = 1) {
+  const MAX_LISTENERS = 100;
+  const redisClient = queue.client;
+  const maxListeners = Math.min(redisClient.getMaxListeners(), MAX_LISTENERS);
+  const currentCounts = queue.eventNames().reduce((counts, eventName) => ({
+    ...counts,
+    [eventName]: queue.listenerCount(eventName),
+  }), {});
+  const totalCount = Object.values(currentCounts).reduce((acc, count) => acc + count, 0);
+  const newListenerCount = totalCount + num;
+  if (newListenerCount > maxListeners) {
+    redisClient.setMaxListeners(newListenerCount);
+  }
+}
```

Comment:
This doesn't do what I suggested. I want the listeners to stop scaling at 100. If we leave it like this, they will still scale infinitely.

Reply:
In fact, if they ever get over 100, "setMaxListeners" will be called every time no matter what, since written this way, maxListeners is the lesser of the current max listeners or 100.

Reply:
I'm having trouble following your comments lets talk in the AM.

Reply:
Was this resolved, should it throw an error

Reply:
This was fixed.